### PR TITLE
Add Module interface and HookCatalog utility

### PR DESCRIPTION
**Pull request by rpanchyk**

### CODE REVIEW

File: src/main/java/org/prebid/server/hooks/v1/Module.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.prebid.server.hooks.v1;
+
+import java.util.Set;
+
+/**
+ * Cares of the module identification among other modules and supplies a collection of available {@Hook}s.
+ */
+public interface Module {
+
+    String code();
```
Comment (muuki88) [inline src/main/java/org/prebid/server/hooks/v1/Module.java:10]: Is the `code` a unique identifier or can there be duplicates?

Comment (muuki88) [reply src/main/java/org/prebid/server/hooks/v1/Module.java:10]:
```suggestion
    /**
     * A unique identifier that should be unique among other available modules.
     */
    String code();
```

Maybe document a few of the usages of this code? Analytic events, logging, referencing in other modules?

Comment (rpanchyk) [reply src/main/java/org/prebid/server/hooks/v1/Module.java:16]: Thanks for pointing, i'll just document what we imaging how it should work so far.

File: src/main/java/org/prebid/server/hooks/v1/Module.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    /**
+     * An identifier that should be unique among other available modules.
+     */
+    String code();
+
+    /**
+     * Collection of hooks available through the module.
+     */
+    Set<Hook> hooks();
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/hooks/v1/Module.java:21]: Change to `Collection` ?

**Review by schernysh — APPROVED** [review]

**Review by DGarbar — APPROVED** [review]

**Pull request merged**